tests/integration/test_catalog.py

```diff
@@ -343,3 +345,67 @@ def test_update_namespace_properties(test_catalog: Catalog, database_name: str)
         else:
             assert k in update_report.removed
     assert "updated test description" == test_catalog.load_namespace_properties(database_name)["comment"]
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize("test_catalog", CATALOGS)
+def test_update_table_schema(test_catalog: Catalog, table_schema_nested: Schema, table_name: str, database_name: str) -> None:
+    identifier = (database_name, table_name)
+    test_catalog.create_namespace(database_name)
+    table = test_catalog.create_table(identifier, table_schema_nested)
+
+    update = table.update_schema().add_column("new_col", LongType())
+    update.commit()
+
+    loaded = test_catalog.load_table(identifier)
+
+    assert loaded.schema().find_field("new_col", case_sensitive=False)
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize("test_catalog", CATALOGS)
+def test_update_table_schema_conflict(
+    test_catalog: Catalog, table_schema_nested: Schema, table_name: str, database_name: str
+) -> None:
+    identifier = (database_name, table_name)
+    test_catalog.create_namespace(database_name)
+    table = test_catalog.create_table(identifier, table_schema_nested)
+
+    update = table.update_schema().add_column("new_col", LongType())
+
+    # update the schema concurrently so that the original update fails
+    concurrent_table = test_catalog.load_table(identifier)
+    # The test schema is assumed to have a `bar` column that can be deleted.
+    concurrent_update = concurrent_table.update_schema(allow_incompatible_changes=True)
+    concurrent_update.set_identifier_fields("foo")
+    concurrent_update.update_column("foo", required=True)
+    concurrent_update.delete_column("bar")
+    concurrent_update.commit()
```

Comment:
I think this is a bit more readable wdyt?
```suggestion
    with concurrent_table.update_schema(allow_incompatible_changes=True) as concurrent_update:
      concurrent_update.set_identifier_fields("foo")
      concurrent_update.update_column("foo", required=True)
      concurrent_update.delete_column("bar")
```

Reply:
+1 on the use of the context-manager :)

Reply:
Great call, fixed!